### ISSUE TITLE
Feature request: Allow moving position of video while playing video

### ISSUE THREAD

**Issue by tex** [kanehekili/VideoCut#42]
Thank you for this project, exactly what I need!

**Comment by kanehekili** [comment]
Not sure why you changed that. Would you mind explaining? Otherwise, thanks a lot for your contribution. 
Note: The slider should be disabled while playing the video or when cutting or when no film has been selected. From my point of view, this is too generic.

**Comment by tex** [comment]
I am recording from DVB-T2, mpeg container, hevc. When I move slider, the preview window looks in 99% cases like this:

![image](https://github.com/user-attachments/assets/e880ad62-955e-4230-a7c5-da72a0a82518)

To get clear picture it needs to be moved by few frames back or forth. But it is tedious to do it manually. Much better to just keep playing which produces good picture almost immediately.

I was testing this couple days and cut many my recordings and all looks fine. Only inconvenience is that the slider moves a bit back every time I move it somewhere when video is playing, but it gets back to my position in like couple hundred milliseconds. So I do not complain much about that.

Anyway, other cutting software or players do not disable slider when they play video... ;-)

Moving slider while there is no video is not crashing application so that is not an issue, no one is going to do that and if it tries it does nothing bad. Not worth complicating it in my opinion. Also I never touched the slider when application was cutting it. But I tried it now and everything looks good, video is exporting and preview is moving. Not problem from user point of view.

Anyway, it was just quick fix for my need. Feel free to improve it or not include it. I am running my copy on my side and I am happy. Again, thank you! Perfect - simple to use application - and written in python so very easy to understand and modify! It is unbelievable that at the end of year 2024 there is no other cutting application that would be lightweight (Yes I know about that one that brings 500MB of code or something), modifiable, simple and working available on Linux.

**Comment by kanehekili** [comment]
OK. that is an artefact by mpvplayer. Your proposed "fix" is a workaround, not a solution. If you could provide a small part of the recording, I might be able to tune the mpv parameters. Note that Videocut has been written primarily for mpegts films - recorded from TV and then to be cut.

**Comment by tex** [comment]
Interesting is that all resulting videos I cut with this tool don't show those artefacts.
I'll try to record some short video and see if it has it.
Anyway, as an user, I like ability to move slider when it is playing video :)

**Comment by tex** [comment]
Hmm, my setup is that I have recordings on server and access them with NFS. This way I am getting those artefacts. When I copy the recording to my local laptop I do not get those artefacts.

But anyway, look at any video player / cutter, they do allow use of slider when playing. It is much nicer to just move slider and see what is there then to move cursor to stop, move it back to slider and then move it back to play.

**Comment by kanehekili** [comment]
Well, the idea is to move to the first I - frame, so you wouldn't have to be bothered with "searching" it. But that info helps me - I'll get some of my TV recordings via NFS or SMB and report back.. In addition I'll check your suggestion of the free slider...

**Comment by tex** [comment]
Also it happens only on recordings that have big quality / screen resolution therefore they are quite big (10G per a recording which is about 4G per movie). On less quality recordings it doesn't happen that much as I think it gets cached more easily. The NFS is connected with not very fast wifi.

**Comment by kanehekili** [comment]
Tested your change and approving it. I will include it in my next release since it is an improvement. Thanks for reporting.

**Comment by tex** [comment]
Thank you for considering the slider behavior!

But still the problem with I-frame is not fixed. I do print the frameNbr and it has always a good and sane values, never bigger that sys.maxsize. So I am wondering what are your thought about using the min and sys.maxsize.

**Comment by kanehekili** [comment]
No, that fix just came while testing a file on an SMB share. I've never seen it before, but it triggered an exception - so better safe than sorry. 
Currently, I'm not able to move to the first I frame in UI - (the remuxer does) since the UI is agnostic concerning the backend like OpenCV and MPV. So I can't provide a solution right now. I expect the backend to do it (which obviously works on local files, on SMB shares, but not on your NFS system)

**Comment by kanehekili** [comment]
You may open another issue, but allowing the slider to be moved while playing has been solved
